internal/pkg/cli/task_run.go

```diff
@@ -259,16 +261,54 @@ func (o *runTaskOpts) Validate() error {
 	return nil
 }
 
+func (o *runTaskOpts) validateRunnerTypeOptions() error {
+	// Subnets cannot be specified with default.
```

Comment:
This is pretty readable thanks to the comments, but I think we could structure it like this and get rid of the comments:
```go 
if err := o.validateFlagsWithDefaultCluster(); err != nil { return err }
if err := o.validateFlagsWithSubnets(); err != nil { return err }
if err := o.validateFlagsWithSecurityGroups(); err != nil { return err }

func (o  *runTaskOpts) validateFlagsWithDefaultCluster() error {
    if !o.useDefaultCluster  {
         return nil
    }
    // put logic
}
```